lib/zipkin-tracer/zipkin_sender_base.rb

```diff
@@ -40,7 +40,7 @@ def start_span(trace_id, name, timestamp = Time.now)
     def skip_flush?(span)
       return true if span.kind == Trace::Span::Kind::CLIENT && span.has_parent_span?
 
-      if span.kind == Trace::Span::Kind::PRODUCER
+      if span.kind.nil? || span.kind == Trace::Span::Kind::PRODUCER
         return true if spans.any? { |s| s.kind == Trace::Span::Kind::SERVER || s.kind == Trace::Span::Kind::CONSUMER }
```

Comment:
I wonder, should we rewrite this method to check return true if any span in spans except the current is s.kind == Trace::Span::Kind::SERVER || s.kind == Trace::Span::Kind::CONSUMER

Wouldn't that be sufficient?

Reply:
how about
```ruby
    def skip_flush?(span)
      case span.kind
      when Trace::Span::Kind::SERVER, Trace::Span::Kind::CONSUMER
        false
      when Trace::Span::Kind::CLIENT
        span.has_parent_span?
      else
        spans.any? { |s| s.kind == Trace::Span::Kind::SERVER || s.kind == Trace::Span::Kind::CONSUMER }
      end
    end
```
?

Reply:
or you mean we don't even need to check `span.has_parent_span?` ?

Reply:
updated in ba42be2